app/forecasts/fetch.py

```diff
@@ -0,0 +1,210 @@
+        # before anything, lets translate the timestamp to utc
+        utc_timestamp = timestamp.astimezone(timezone.utc)
+        if self.current_timestamp != utc_timestamp:
+            # Swap out current with previous
+            self.prev_timestamp = self.current_timestamp
+            self.prev_values = self.current_values
+            # Reset current
+            self.current_values = {}
+            self.current_timestamp = utc_timestamp
+
+        self.current_values[key] = value
+
+    def is_before_noon(self, timestamp: datetime.datetime):
+        """ Return true if given time is before utc noon """
+        return timestamp.time() < self.utc_noon
+
+    def is_after_noon(self, timestamp: datetime.datetime):
+        """ Retrun true if given time is after uct noon """
+        return timestamp.time() > self.utc_noon
+
+    def calculate_noon_value(self) -> WeatherForecastValues:
+        """ Calcualte the interpolated noon value (if possible) """
+        # If the previous timestamp was before noon, and the current timestamp is after noon,
+        # it means there is no noon value, and we can interpolate one.
+        result = None
+        if (self.prev_timestamp and
+                self.is_before_noon(self.prev_timestamp) and
+                self.is_after_noon(self.current_timestamp)):
+            noon = datetime.datetime(year=self.prev_timestamp.year, month=self.prev_timestamp.month,
+                                     day=self.prev_timestamp.day, hour=self.utc_noon.hour,
+                                     minute=self.utc_noon.minute, tzinfo=timezone.utc)
+            result = WeatherForecastValues(datetime=noon)
+            # x-axis is the timestamp
+            x_axis = (self.prev_timestamp.timestamp(),
+                      self.current_timestamp.timestamp())
+            # for each key value, we have a y-axis
+            y_axis = {}
+            for key, value in self.prev_values.items():
+                y_axis[key] = [value, self.current_values[key]]
+            for key, value in y_axis.items():
+                function = interp1d(x_axis, value, kind='linear')
+                interpolated_value = function(noon.timestamp())
+                # the interpolated value is in the form of an array, we just want the 1st element.
+                setattr(result, key, interpolated_value.item(0))
+        return result
+
+
+def _get_stations_for_polygon(stations, polygon: Polygon) -> List:
+    """ There could be multiple stations in this polygon, return all that match.
+    """
+    stations_in_polygon = []
+    for station in stations:
+        point = Point(station.long, station.lat)
+        logger.info('testing if %s is within %s', point, polygon)
+        if point.within(polygon):
+            stations_in_polygon.append(station)
+    return stations_in_polygon
+
+
+def _add_prediction_to_forecast(forecast: WeatherForecast, prediction: Prediction, points: List[float],
```

Comment:
missing docstring - it's not immediately clear what the diff is btwn a Prediction & a WeatherForecast

Reply:
Adding the doc-string is easily done.  The naming gets me going in circles though. I'm not sure what a "good" name for the database model is - maybe "ModelPrediction" is a better name? Then it becomes
"_add_model_prediction_to_" - and then the next question is that schema object that's currently called WeatherForecast - but is actually really specific to a ModelForecast.... so really it should become "_add_model_prediction_to_model_forecast" - and a bunch of refactoring to go with it.
Thoughts?